### PR TITLE
Enable master link replacement feature for cpp

### PR DESCRIPTION
**Pull request by sima-zhu**
This is a follow up step of master link replacement work.

Tested in template:
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=532011&view=logs&j=f729e1db-2340-5697-6505-4f6bf41660db&t=926666bf-069f-5669-9149-9ad56a04b65b

Retrieved release tag successfully:
image

Github IO: https://azuresdkdocs.blob.core.windows.net/$web/cpp/azure-template/1.0.0-beta.3/index.html

### CODE REVIEW

File: sdk/template/azure-template/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Release History
 
-## 1.0.0-beta.1 (Unreleased)
+## 1.0.0-beta.2 (2020-09-09)
+
+* Enable master replacement link feature for cpp.
```
Comment (joshfree) [inline sdk/template/azure-template/CHANGELOG.md:5]: Could this line say something about "Engineering System" so external customers know its not an SDK feature

**Review by joshfree — APPROVED** [review]

**Review by weshaggard — APPROVED** [review]

File: sdk/template/azure-template/CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,14 @@
 # Release History
+## 1.0.0-beta.4 (Unreleased)
+
+
+## 1.0.0-beta.3 (2020-09-09)
+
+* (Engineer System) Enable master replacement link feature for cpp, attempt 2
```
Comment (ahsonkhan) [inline sdk/template/azure-template/CHANGELOG.md:7]:
```suggestion
* (Engineering System) Enable master replacement link feature for cpp, attempt 2
```

File: sdk/template/azure-template/CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,14 @@
...
+
+## 1.0.0-beta.2 (2020-09-09)
+
+* (Engineer System) Enable master replacement link feature for cpp, attempt 1
```
Comment (ahsonkhan) [inline sdk/template/azure-template/CHANGELOG.md:11]:
```suggestion
* (Engineering System) Enable master replacement link feature for cpp, attempt 1
```

**Pull request merged**